### PR TITLE
[SYNCOPE-1656] remediation now is generated also on pull update failure

### PR DESCRIPTION
**Pull request by andrea-patricelli**
Also made some refactoring to `AbstractPullResultHandler` and `DefaultUserPullResultHandler`

### CODE REVIEW

**Comment by ilgrosso** [comment]
@andrea-patricelli please remove all formatting-only changes, to render clear the actual modifications you are submitting, tjx

File: fit/core-reference/src/test/java/org/apache/syncope/fit/AbstractITCase.java [hunk 1]
```diff
@@ -702,4 +767,9 @@ protected void removeLdapRemoteObject(
         } while (results.isEmpty() && i < maxWaitSeconds);
         return results;
     }
+
+    protected void cleanUpRemediations() {
```
Comment (ilgrosso) [inline fit/core-reference/src/test/java/org/apache/syncope/fit/AbstractITCase.java:771]: This method seems to be used only within `PullTaskITCase`: can you move it to that class?

File: fit/core-reference/src/test/java/org/apache/syncope/fit/AbstractITCase.java [hunk 2]
```diff
@@ -635,6 +668,38 @@ protected void updateLdapRemoteObject(
         }
     }
 
+    protected Pair<String, Set<Attribute>> prepareLdapAttributes(
```
Comment (ilgrosso) [inline fit/core-reference/src/test/java/org/apache/syncope/fit/AbstractITCase.java:671]: This method is used only by a single test, please move it back to the containing test class

**Review by ilgrosso — CHANGES_REQUESTED** [review]

**Comment by ilgrosso** [comment]
@andrea-patricelli watch out that the new test case has been failing during the previous 2 runs

**Comment by andrea-patricelli** [comment]
> @andrea-patricelli watch out that the new test case has been failing during the previous 2 runs
> 
> See:
> 
> * https://github.com/apache/syncope/actions/runs/1697628191
> * https://github.com/apache/syncope/actions/runs/1697471169
> 
> ```
> PullTaskITCase.issueSYNCOPE1656:1457 expected: <1> but was: <5>
> ```

I guess that there are some data created by other tests that are "invalidating" this one. Going to fix.

**Review by ilgrosso — APPROVED** [review]
LGTM please go ahead, squash and merge

**Pull request merged**